src/wallet/rpcwallet.cpp

```diff
@@ -583,6 +583,57 @@ static UniValue signmessage(const JSONRPCRequest& request)
     return EncodeBase64(vchSig.data(), vchSig.size());
 }
 
+static CAmount GetReceived(interfaces::Chain::Lock& locked_chain, CWallet * const wallet, const UniValue& params, bool by_label) EXCLUSIVE_LOCKS_REQUIRED(wallet->cs_wallet)
+{
+    std::set<CTxDestination> addresses;
+
+    if (by_label) {
+        // Get the set of pub keys assigned to label
+        std::string label = LabelFromValue(params[0]);
+        addresses = wallet->GetLabelAddresses(label);
+    } else {
+        // Get the address
+        CTxDestination dest = DecodeDestination(params[0].get_str());
+        if (!IsValidDestination(dest)) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Bitcoin address");
+        }
+        CScript script_pub_key = GetScriptForDestination(dest);
+        if (!wallet->IsMine(script_pub_key)) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Address not found in wallet");
+        }
+        addresses.insert(dest);
+    }
+
+    // Minimum confirmations
+    int min_depth = 1;
+    if (!params[1].isNull())
+        min_depth = params[1].get_int();
+
+    // Tally
+    CAmount amount = 0;
+    for (const std::pair<const uint256, CWalletTx>& wtx_pair : wallet->mapWallet) {
+        const CWalletTx& wtx = wtx_pair.second;
+        if (wtx.IsCoinBase() || !locked_chain.checkFinalTx(*wtx.tx)) {
+            continue;
+        }
+
+        int depth = wtx.GetDepthInMainChain();
+        if (depth < min_depth) {
+            continue;
+        }
```

Comment:
More a matter of taste, but this could simply be put in the `if` condition above and without and extra variable (that is used only once anyway)? Like
```
if (wtx.IsCoinBase() || !locked_chain.checkFinalTx(*wtx.tx) || wtx.GetDepthInMainChain() < min_depth) {
    continue;
}
```